### ISSUE TITLE
make config-galaxy-s2

### ISSUE THREAD

**Issue by matthewp** [andreasgal/B2G#138]
Brand new Galaxy S2, never even had a sim card in it.

[matthew@myhost B2G]$ make config-galaxy-s2
Extracting binary blobs from device, which should be plugged in! ...
Warning, your device has unknown firmware GINGERBREAD.UVKL1
Pulling "libActionShot.so"
3836 KB/s (634545 bytes in 0.161s)
Pulling "libakm.so"
773 KB/s (38380 bytes in 0.048s)
Pulling "libarccamera.so"
4664 KB/s (2578783 bytes in 0.539s)
Pulling "libcamera_client.so"
924 KB/s (90504 bytes in 0.095s)
Pulling "libcameraservice.so"
1105 KB/s (59352 bytes in 0.052s)
Pulling "libcamera.so"
1366 KB/s (147848 bytes in 0.105s)
Pulling "libcaps.so"
3415 KB/s (905356 bytes in 0.258s)
Pulling "libexif.so"
467 KB/s (40968 bytes in 0.085s)
Pulling "libfimc.so"
remote object '/system/lib/libfimc.so' does not exist
Failed to pull libfimc.so. Giving up.
make: **\* [config-galaxy-s2] Error 255

**Comment by fabricedesre** [comment]
You have a firmware we've never seen yet:

|Warning, your device has unknown firmware GINGERBREAD.UVKL1|

$b2g/glue/gonk/device/samsung/galaxys2/extract-files.sh is where we manage that for galaxy S2s

**Comment by matthewp** [comment]
The internets suggests that this firmware was just pushed out recently. I'm going to poke around that script.

**Comment by matthewp** [comment]
So it appears that my firmware is quite a bit different. What is the best strategy, remove the files I don't have, or should I also add ones I do have that aren't listed (will be quite a lot). I guess I'm asking if these libs are supposed to be a comprehensive list or not.

**Comment by jlebar** [comment]
Did you figure this out?

**Comment by matthewp** [comment]
No, I sort of gave up. This firmware is for the T-Mobile Galaxy S2 which is not even supported by Cyanogen at this time. I was able to get the script to pull the correct files, but ran into other errors. Since there are android hackers working on porting Cyanogen to this device I would wait on them to do the work that they are good at.

**Comment by shianyow** [comment]
@matthewp I am working on AT&T Galaxy S2 Skyrocket, which should be same hardware as T-Mobile version.
If you are still waiting for it and want to see B2G running, just let me know.

**Comment by matthewp** [comment]
@shianyow  Why do you think they are the same? My understanding from the research I did at the time is that the T-Mobile is quite a bit different from all of the other versions, hence why it is not supported by CyanogenMod (which B2G uses code from).  Unless that has changed, and I don't think it has, this phone is probably a lost cause.

**Comment by shianyow** [comment]
@matthewp Ah, I should say AT&T SGH-I727 and T-Mobile SGH-T989 are 95% identical in hardware (Qualcomm chip).  But a lot different from international SGS2 I-9100 (Samsung chip).
Here is a good reference to build kernel and flash for these two devices, FYI.
http://forum.xda-developers.com/showthread.php?t=1516051

**Comment by matthewp** [comment]
So, were you able to build B2G for I727?  I would possibly be interested in working on the 5% if someone else has done the 95% :)  I'm not really a kernel hacker so my previous attempts were a bit outside of my wheelhouse.

**Comment by shianyow** [comment]
This is branch to build B2G for I727.
https://github.com/shianyow/B2G/tree/skyrocket

For T989, a different kernel from Samsung should be used.  Here is the repo you can use.
https://github.com/shianyow/kernel-android-galaxy-s2-t989

And you need to create device configuration tree for T989, by reference I727.  Ideally you just need to change the model name.

**Comment by matthewp** [comment]
Thanks @shianyow.  I returned my S2 after I failed to get B2G built for it.  I'm tempted to get another one and give it another go, but I think I better wait and see if someone else can get the T989 to work rather than pay $600 to possibly fail again. Or hopefully some developer phones of the LG models will be available soon. Thanks anyways and great work.

**Comment by cgaspard** [comment]
Hi shianyow I need some help w/ B2G on my i727.   I've done a little bit of debugging and I've found that the script create_odin_img.sh is looking for zImage in a folder that the kernel does not compile into by default.

When I compiled with "make kernel", the resulting zImage file was in B2G/boot/kernel-android-skyrocket/arch/arm/boot/zImage not B2G/glue/gonk/out/target/product/skyrocket/obj/KERNEL_OBJ/arch/arm/boot/zImage

As a work around, I manually copied the file to the expected path.

Now I'm having another issue.   I'm not getting the system.img file same script is expecting to be in the  B2G/glue/gonk/out/target/product/skyrocket/system.img path.  Can you tell me how this file is to be created?   So far I haven't seen anything in the make scripts that create the system.img.

Thanks in advance.

**Comment by shianyow** [comment]
Hi @cgaspard 

The make procedure suppose to be:
$ make config-galaxy-s2-skyrocket
$ make gonk
$ make

Did you see any error during make?
If no make error, it's supposed generate Image and system.img as expected in create_odin_img.sh.

**Comment by cgaspard** [comment]
@shianyow

Success!!

I followed your build instructions instead of the Galaxy S2 ones and I was able to build and install the Odin image.

Only difference was I used make config-skyrocket in the first step.

I'm going to play with the phone tomorrow to see all whats working and whats not.   Is there anything in particular you need help with testing?

**Comment by shianyow** [comment]
Good!  One known issue is no voice during phone call.

**Comment by cgaspard** [comment]
@shianyow 

Okay I was able to play around with the phone today, and I think I may have done something wrong.

First thing I noticed when booting the phone is that it flashes the T-Mobile boot screen.   Which makes me think that maybe I have the wrong device kernel.   I have the ATT i727.

Once B2G boots and Gaia starts up, I do get the lock screen, and the following items work.   Swiping the lock screen, the clock on the lock screen, wifi appears to be working, volume up and down, the power button.

What appears to not be working is the sim card detection, and also Gaia itself doesn't show anything after I swipe the lock screen.

I'm planning on doing some more debugging this evening, but I was curious if you might have any ideas?

Thanks

**Comment by shianyow** [comment]
Hi @cgaspard 

Yes, I encounter T-Mobile boot screen instead of AT&T too.
I was reference below guide to make the create_odin_img.sh script:
http://forum.xda-developers.com/showthread.php?t=1516051
Although it's written for T-Mobile model (t989), but the procedure is same for both devices.  And the kernel source is for i727.  I guess some parameters in the script should be modified to show ATT boot screen, but no idea about this.

I didn't encounter the Gaia issue you mentioned, my Gaia commit id is "5dd83cf0...", which is coming from the B2G skyrocket branch.

**Comment by cgaspard** [comment]
I had to run.

$make install-gaia

That seems to have fixed it.
